lib/mapObjects/MiscObjects.h

```diff
@@ -459,10 +461,22 @@ class DLL_LINKAGE HillFort : public CGObjectInstance, public ICreatureUpgrader
 public:
 	using CGObjectInstance::CGObjectInstance;
 
+	const std::string & getDescriptionToolTip() const
+	{
+		return descriptionToolTip;
+	}
+
+	const std::string & getUnavailableUpgradeMessage() const
+	{
+		return unavailableUpgradeMessage;
+	}
+
 	template <typename Handler> void serialize(Handler &h)
 	{
 		h & static_cast<CGObjectInstance&>(*this);
 		h & upgradeCostPercentage;
+		h & descriptionToolTip;
+		h & unavailableUpgradeMessage;
```

Comment:
This would break loading of 1.5 saves.

Perhaps instead of storing strings, HillFort should return string ID generated in runtime? 
Something like this:
`return TextIdentifier(getHandler()->getBaseTextID(), "description").get();`